File: app/src/main/java/com/kjipo/eartraining/svg/SequenceToSvg.java
package com.kjipo.eartraining.svg;


import com.kjipo.eartraining.data.Note;

import java.util.List;


public class SequenceToSvg {

    private final Glyphs glyphs;


    public SequenceToSvg(Glyphs glyphs) {
        this.glyphs = glyphs;
    }


    public void transformToSvg(List<Note> temporalElements, SvgSequenceConfig svgSequenceConfig) {
        int barWidth = svgSequenceConfig.getBarWidth();

        // TODO


//        ("file:///android_asset/js/glyphs.json");

//        Gson gson = new Gson();
//        gson.fromJson()


        for (Note temporalElement : temporalElements) {
            // TODO Just for testing
            drawPathAt(1, glyphs.getPath(temporalElement.getElementType()));


        }


    }


    private void drawPathAt(int xcoord, String path) {
        // TODO

    }


}

File: app/src/main/java/com/kjipo/eartraining/svg/SequenceToSvg.kt
package com.kjipo.eartraining.svg


import android.util.Log
import com.kjipo.eartraining.data.Sequence
import com.kjipo.svg.*


class SequenceToSvg {


    fun transformToSvg(sequence: Sequence?): RenderingSequence {
        if (sequence == null) {
            return RenderingSequence(emptyList(), ViewBox(0, 0, 0, 0))
        }

        Log.i("SVG", "Notes: " + sequence.notes)

        // TODO Only here for testing

        return createScore().score {
            bar {
                clef = Clef.G
                timeSignature = TimeSignature(4, 4)

                note {
                    note = NoteType.A
                    duration = Duration.QUARTER
                }

                note {
                    note = NoteType.H
                    duration = Duration.HALF
                    beamGroup = 1
                }

                note {
                    note = NoteType.C
                    octave = 7
                    duration = Duration.HALF
                }


            }

        }


    }
}
